src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -604,14 +707,37 @@ private function getDefinitionSchema(bool $v3, string $resourceClass, ResourceMe
         }
 
         $options = isset($serializerContext[AbstractNormalizer::GROUPS]) ? ['serializer_groups' => $serializerContext[AbstractNormalizer::GROUPS]] : [];
+
+        if ($v3) {
+            if ('application/vnd.api+json' === $mimeType) {
```

Comment:
Cab we extract the code generating the JSON Schema corresponding to every supported formats in specific classes. It will make this class less complex (it is already to big and complex), and will allow users to register their on JSON Schema generators for their custom formats.

Reply:
Yeah, that's on my todo list. I'll add a todo list to the so we know what needs to happen.

Reply:
I added some functionality to do this, its not finished still needs to get the formatters from the config and create more functions for more complex setups. Its a start, just let me know if its going in the correct direction.